packages/@aws-cdk/aws-servicediscovery/lib/service.ts

```diff
@@ -0,0 +1,224 @@
+   */
+  namespace: INamespace;
+
+  /**
+   * The DNS type of the record that you want AWS Cloud Map to create. Supported record types
+   * include A, AAAA, A_AAAA, CNAME, and SRV.
+   */
+  dnsType: DnsRecordType;
+
+  /**
+   * The amount of time, in seconds, that you want DNS resolvers to cache the settings for this
+   * record.
+   *
+   * @default 60
+   */
+  dnsTtlSec?: number;
+
+  /**
+   * Public DNS namespaces only. A complex type that contains settings for an optional health check.
+   * If you specify settings for a health check, AWS Cloud Map associates the health check with the
+   * records that you specify in DnsConfig.
+   */
+  healthCheckConfig?: HealthCheckConfig;
+
+  /**
+   * Structure containing failure threshold for a custom health checker.
+   * See: https://docs.aws.amazon.com/cloud-map/latest/api/API_HealthCheckCustomConfig.html
+   */
+  healthCheckCustomConfig?: HealthCheckCustomConfig;
+
+  /**
+   * The routing policy that you want to apply to all DNS records that AWS Cloud Map creates when you
+   * register an instance and specify this service.
+   *
+   * @default WEIGHTED for CNAME records, MULTIVALUE otherwise
+   */
+  routingPolicy?: RoutingPolicy;
+
+  /**
+   * A name for the Service.
+   *
+   * @default CloudFormation-generated name
+   */
+  name?: string;
+
+  /**
+   * A description of the service.
+   *
+   * @default none
+   */
+  description?: string;
+}
+
+/**
+ * Define a CloudMap Service
+ */
+export class Service extends cdk.Construct {
+  /**
+   * A name for the Cloudmap Service.
+   */
+  public readonly name: string;
+
+  public readonly namespaceId: string;
+
+  /**
+   *  The namespace for the Cloudmap Service.
+   */
+  public readonly namespace: INamespace;
+
+  /**
+   * The ID of the namespace that you want to use for DNS configuration.
+   */
+  public readonly serviceArn: string;
+
+  constructor(scope: cdk.Construct, id: string, props: ServiceProps) {
+    super(scope, id);
+
+    const routingPolicy = props.dnsType && props.dnsType === DnsRecordType.Cname
+      ? RoutingPolicy.Weighted
+      : RoutingPolicy.Multivalue;
+
+    const dnsType = props.dnsType !== undefined ? props.dnsType : DnsRecordType.A;
+
+    const service = new CfnService(this, 'Resource', {
+      name: props.name,
+      description: props.description,
+      dnsConfig: props.namespace.type === NamespaceType.Http ? undefined :
+      {
+        dnsRecords: [
+          {
+            type: dnsType,
+            ttl: props.dnsTtlSec !== undefined ? props.dnsTtlSec.toString() : '60',
+          }
+        ],
+        namespaceId: props.namespace.namespaceId,
+        routingPolicy,
+      },
+      healthCheckCustomConfig: props.healthCheckCustomConfig,
+    });
+
+    this.name = service.serviceName;
+    this.serviceArn = service.serviceArn;
+    this.namespace = props.namespace;
+  }
+}
+
+export interface DnsConfig {
+  dnsRecords: DnsRecord[],
+    namespaceId: string,
```

Comment:
Not really a nice interface to `export`. If it's a reference to some other resource, we don't reference it by string-id by by interface (`INamespace` here)

Reply:
Hm, I was only using it to make it easier in the constructor (line 97), but perhaps exporting the interface this way isn't the way to go?